[extension/dbstorage] Failing test TestExtensionIntegrityWithPostgres

### Component(s)

extension/storage/dbstorage

### Describe the issue you're reporting

[Failing CI/CD action](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/12657427377/job/35272224641#step:9:606)

Failure output:
```
=== FAIL: . TestExtensionIntegrityWithPostgres (7.44s)
    extension_test.go:45: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/storage/dbstorage/extension_test.go:45
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/storage/dbstorage/extension_test.go:39
        	Error:      	Received unexpected error:
        	            	failed to connect to `user=root database=db`: 127.0.0.1:32769 (127.0.0.1): failed to receive message: read tcp 127.0.0.1:51264->127.0.0.1:32769: read: connection reset by peer
        	Test:       	TestExtensionIntegrityWithPostgres
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x723b12]
```

--- a/extension/storage/dbstorage/extension_test.go
+++ b/extension/storage/dbstorage/extension_test.go
@@ -12,6 +12,7 @@ import (
 	"runtime"
 	"sync"
 	"testing"
+	"time"
 
 	ctypes "github.com/docker/docker/api/types/container"
 	"github.com/docker/go-connections/nat"
@@ -48,6 +49,10 @@ func TestExtensionIntegrityWithPostgres(t *testing.T) {
 		}
 	})
 	require.NoError(t, err)
+
+	// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37079
+	// DB instantiation fails if we instantly try to connect to it, wait for 10s before starting the extension
+	time.Sleep(10 * time.Second)
 
 	testExtensionIntegrity(t, se)
 }

